cirq-core/cirq/circuits/circuit.py

```diff
@@ -1041,89 +1041,86 @@ def _superoperator_(self) -> np.ndarray:
             circuit_superoperator = moment_superoperator @ circuit_superoperator
         return circuit_superoperator
 
+    @_compat.deprecated_parameter(
+        deadline='v0.16',
+        fix='Inject identity operators to include untouched qubits.',
+        parameter_desc='qubits_that_should_be_present',
+        match=lambda args, kwargs: 'qubits_that_should_be_present' in kwargs,
+    )
+    @_compat.deprecated_parameter(
+        deadline='v0.16',
+        fix='Only use keyword arguments.',
+        parameter_desc='positional args',
+        match=lambda args, kwargs: len(args) > 1,
+    )
     def final_state_vector(
         self,
+        # TODO(v0.16): Force kwargs and match order found in:
+        # cirq-core/cirq/sim/mux.py:final_state_vector
         initial_state: 'cirq.STATE_VECTOR_LIKE' = 0,
         qubit_order: 'cirq.QubitOrderOrList' = ops.QubitOrder.DEFAULT,
         qubits_that_should_be_present: Iterable['cirq.Qid'] = (),
         ignore_terminal_measurements: bool = True,
-        dtype: Type[np.number] = np.complex128,
+        dtype: Type[np.number] = np.complex64,
```

Comment:
This looks like a breaking change, because user code relying on `circuit.final_state_vector()` assuming default `dtype=np.complex128`, can now break due to the reduced precision ? Do you think we need to go through a deprecation cycle here?

Reply:
Done. Also made `ignore_terminal_measurements` default to `None` to better capture use-of-default, and updated call sites to avoid deprecated code.

Reply:
Okay, maybe not _all_ of the call sites. Working through the failing CI tests now.